azerothcore/azerothcore-wotlk
# fix(Scripts/BlackrockSpire): Rend Blackhand improvements

## Changes Proposed:
-  DEV: Implement two new methods to summonlist - IsAnyCreatureAlive() & IsAnyCreatureInCombat(), self-explanatory
-  Reset Rend's encounter if any of the adds enters evade mode, despawns AND no other of his summons is in combat
- Implemented missing emote when Rend dismounts & Nefarian's text on wipe
- Adds despawn after being out of combat 
- Waves are no longer timed, instead they start 10 seconds after the last summon is killed
- Moved all of this summons to database

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8731

## SOURCE:


## Tests Performed:

- Tested in-game.
- 


## How to Test the Changes:


1.go creature id 10429
2. Try the things described in *changes proposed* & see if they address the issues reported
3.

## Known Issues and TODO List:


- There are still improvements to be made, but not enough resources (texts, f.e).


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
